test/reply-error.test.js

```diff
@@ -444,6 +444,60 @@ test('should throw an error if the custom serializer does not serialize the payl
   })
 })
 
+test('default error handling should set headers', t => {
+  t.plan(4)
+
+  const fastify = Fastify()
+  fastify.get('/', function (req, reply) {
+    const err = new Error('kaboom')
+    err.headers = {
+      'fake-random-header': 'abc'
+    }
+    reply.send(err)
+  })
+
+  fastify.inject({
+    method: 'GET',
+    url: '/'
+  }, (err, res) => {
+    t.error(err)
+    t.strictEqual(res.statusCode, 500)
+    t.strictEqual(res.headers['fake-random-header'], 'abc')
```

Comment:
I think that a check for `Content-Length` here is good as well.

Reply:
Added the content-length check